### PR TITLE
Guard agains post-processing issues around package upload.

### PR DESCRIPTION
**Pull request by isoos**
This is to cover the morning's issue with the pana publishing.

I don't think any of the steps warrant a retry (e.g. Jobs will be created eventually even if they fail here), a bulk try-catch should shield the users to the internals of the post-processing.

### CODE REVIEW

File: app/lib/frontend/backend.dart [hunk 1]
```diff
@@ -522,30 +522,35 @@ class GCloudPackageRepository extends PackageRepository {
...
-        .triggerAnalysis(newVersion.package, newVersion.version, <String>{});
-    await dartdocClient
-        .triggerDartdoc(newVersion.package, newVersion.version, <String>{});
+      // Trigger analysis and dartdoc generation. Dependent packages can be left
+      // out here, because the dependency graph's background polling will pick up
+      // the new upload, and will trigger analysis for the dependent packages.
+      await analyzerClient
+          .triggerAnalysis(newVersion.package, newVersion.version, <String>{});
+      await dartdocClient
+          .triggerDartdoc(newVersion.package, newVersion.version, <String>{});
```
Comment (jonasfj) [inline app/lib/frontend/backend.dart]: Why not trigger these concurrently?

Comment (jonasfj) [reply app/lib/frontend/backend.dart]: We could even do emailing concurrently with this...

File: app/lib/frontend/backend.dart [hunk 2]
```diff
@@ -522,30 +522,35 @@ class GCloudPackageRepository extends PackageRepository {
...
 
-    if (finishCallback != null) {
-      await finishCallback(newVersion);
+      if (finishCallback != null) {
+        await finishCallback(newVersion);
```
Comment (jonasfj) [inline app/lib/frontend/backend.dart]: Please document in a single line what this is used for?

Comment (isoos) [reply app/lib/frontend/backend.dart]: I think we should probably remove this, it is no longer used. I'm not sure why we had it in the first place, but after a while we were using it for triggering the analysis, and then after a refactor we've got a new place for that.
Should I do this in the PR?

File: app/lib/frontend/backend.dart (hunk 1 above)
Comment (isoos) [reply app/lib/frontend/backend.dart]: updated to do that.

File: app/lib/frontend/backend.dart (hunk 2 above)
Comment (jonasfj) [reply app/lib/frontend/backend.dart]: I filed https://github.com/dart-lang/pub-dev/issues/2304

**Pull request merged**